api/src/pcapi/routes/native/v1/offers.py

```diff
@@ -42,7 +43,38 @@ def get_offer(offer_id: str) -> serializers.OfferResponse:
         )
         .options(joinedload(Offer.venue).joinedload(Venue.googlePlacesInfo))
         .options(joinedload(Offer.mediations))
-        .options(joinedload(Offer.product).load_only(Product.id, Product.thumbCount))
+        .options(
+            joinedload(Offer.product).load_only(Product.id, Product.thumbCount).joinedload(Product.productMediations)
+        )
+        .outerjoin(Offer.lastProvider)
+        .options(sa.orm.contains_eager(Offer.lastProvider).load_only(Provider.localClass))
+        .filter(Offer.id == offer_id, Offer.validation == OfferValidationStatus.APPROVED)
+        .first_or_404()
+    )
+
+    if offer.isActive and providers_repository.is_cinema_external_ticket_applicable(offer):
+        api.update_stock_quantity_to_match_cinema_venue_provider_remaining_places(offer)
+
+    return serializers.OfferResponse.from_orm(offer)  # type: ignore [return-value]
```

Comment:
Pour éviter cette erreur mypy, tu peux typer plus génériquement la classe BaseOfferResponse

Exemple qui passe mypy (`S` est un mauvais nom 😅 mais ça montre l'idée)
Je me suis inspiré de [cette issue sur SO](https://stackoverflow.com/questions/63995525/mypy-updating-return-value-from-child-method-using-childs-name-instead-of-paren)

![image](https://github.com/pass-culture/pass-culture-main/assets/14235661/2ba3636f-b4a7-4e21-a6a3-10e06849fdc7)